### DISCUSSION TITLE
Need Help with configurating snipeIT local connection

### DISCUSSION THREAD

**Discussion by J1MP123** [snipe/snipe-it#11037] [Q&A]
Good evening,

I'm trying to configure snipeIT connection for localhost accessing. but each time it displays 
![image](https://user-images.githubusercontent.com/93220034/167269485-63921114-3aa4-43d1-a93c-c0b62fb06fb0.png)

snipeIT works on the host itself but not accessible from any other local device.

**WAMP-V3.2.6**
**Windows 10 Wamp installation.**

**My httpd-vhosts.confg** 

```
# Virtual Hosts
#
<VirtualHost *:80>
  ServerName localhost
  DocumentRoot "${INSTALL_DIR}/www/snipe-it/public"
  <Directory "${INSTALL_DIR}/www">
    Options +Indexes +Includes +FollowSymLinks +MultiViews
    AllowOverride All
	Require all granted
  </Directory>
</VirtualHost>
```
httpd.conf changes
`
ServerName 172.16.0.115:80
<Directory />
    AllowOverride All
	Require all granted
</Directory>`
**Appache error log**
`[Sat May 07 20:13:49.816911 2022] [mpm_winnt:notice] [pid 11212:tid 736] AH00422: Parent: Received shutdown signal -- Shutting down the server.
[Sat May 07 20:13:51.848128 2022] [mpm_winnt:notice] [pid 9748:tid 712] AH00364: Child: All worker threads have exited.
[Sat May 07 20:13:53.395022 2022] [mpm_winnt:notice] [pid 11212:tid 736] AH00430: Parent: Child process 9748 exited successfully.
[Sat May 07 20:13:54.957481 2022] [mpm_winnt:notice] [pid 10520:tid 688] AH00455: Apache/2.4.51 (Win64) PHP/7.4.26 configured -- resuming normal operations
[Sat May 07 20:13:54.957481 2022] [mpm_winnt:notice] [pid 10520:tid 688] AH00456: Apache Lounge VS16 Server built: Oct  7 2021 16:27:02
[Sat May 07 20:13:54.957481 2022] [core:notice] [pid 10520:tid 688] AH00094: Command line: 'c:\\wamp64\\bin\\apache\\apache2.4.51\\bin\\httpd.exe -d C:/wamp64/bin/apache/apache2.4.51'
[Sat May 07 20:13:54.957481 2022] [mpm_winnt:notice] [pid 10520:tid 688] AH00418: Parent: Created child process 3956
[Sat May 07 20:13:55.660969 2022] [mpm_winnt:notice] [pid 3956:tid 688] AH00354: Child: Starting 64 worker threads.
[Sat May 07 20:34:00.573447 2022] [authz_core:error] [pid 3956:tid 1192] [client 172.16.0.108:55457] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/
[Sat May 07 20:34:00.729629 2022] [authz_core:error] [pid 3956:tid 1192] [client 172.16.0.108:55457] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/favicon.ico, referer: http://172.16.0.115/
[Sat May 07 20:44:32.990230 2022] [authz_core:error] [pid 3956:tid 1212] [client 172.16.0.108:55499] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/
[Sat May 07 20:44:33.349523 2022] [authz_core:error] [pid 3956:tid 1212] [client 172.16.0.108:55499] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/favicon.ico, referer: http://172.16.0.115/`

Any ideas? What am i missing?

**Comment by uberbrady** [comment]
localhost is inherently only accessible from the same machine that it is on. When you're trying to access localhost from other machines, you're accessing that machine itself.

**Reply by J1MP123** [reply]
Hello Uberbrady Changed :
`# Virtual Hosts
#
<VirtualHost *:80>
  ServerName 172.16.0.115
  DocumentRoot "${INSTALL_DIR}/www/snipe-it/public"
  <Directory "${INSTALL_DIR}/www">
    Options +Indexes +Includes +FollowSymLinks +MultiViews
    AllowOverride All
	Require all granted
  </Directory>
</VirtualHost>
`
and httpd.conf changed to
https://pastebin.com/yFwuDP6N

any ideas why it's still getting permissions denied?
![image](https://user-images.githubusercontent.com/93220034/167696037-37e46254-0e4a-46db-8b7f-6770b4354ae8.png)

**Reply by J1MP123** [reply]
Should i change .env file for snipeit configuration if my ip range is 172.16.0.1?

`# --------------------------------------------
# OPTIONAL: SECURITY HEADER SETTINGS
# --------------------------------------------
APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1
ALLOW_IFRAMING=false
REFERRER_POLICY=same-origin
ENABLE_CSP=false
CORS_ALLOWED_ORIGINS=null
ENABLE_HSTS=false`

**Reply by J1MP123** [reply]
Should i change .env file for snipeit configuration if my ip range is 172.16.0.1?

`# --------------------------------------------
# OPTIONAL: SECURITY HEADER SETTINGS
# --------------------------------------------
APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1
ALLOW_IFRAMING=false
REFERRER_POLICY=same-origin
ENABLE_CSP=false
CORS_ALLOWED_ORIGINS=null
ENABLE_HSTS=false`

**Comment by uberbrady** [comment]
I mean, I'd guess set your `APP_URL` to http://172.16.0.115 to start with - then check your apache error logs for any further clues?

**Reply by J1MP123** [reply]
It's already set to 
`# --------------------------------------------
# REQUIRED: BASIC APP SETTINGS
# --------------------------------------------
APP_ENV=production
APP_DEBUG=false
APP_KEY=base64:OYXHTonG6PJVKYPkl1
APP_URL=http://172.16.0.115/
APP_TIMEZONE='Europe/London'
APP_LOCALE=en-GB
MAX_RESULTS=500`

**Reply by J1MP123** [reply]
[Tue May 10 20:57:11.648883 2022] [core:warn] [pid 4464:tid 636] AH00098: pid file C:/wamp64/bin/apache/apache2.4.51/logs/httpd.pid overwritten -- Unclean shutdown of previous Apache run?
[Tue May 10 20:57:12.523829 2022] [mpm_winnt:notice] [pid 4464:tid 636] AH00455: Apache/2.4.51 (Win64) PHP/7.4.26 configured -- resuming normal operations
[Tue May 10 20:57:12.523829 2022] [mpm_winnt:notice] [pid 4464:tid 636] AH00456: Apache Lounge VS16 Server built: Oct  7 2021 16:27:02
[Tue May 10 20:57:12.523829 2022] [core:notice] [pid 4464:tid 636] AH00094: Command line: 'c:\\wamp64\\bin\\apache\\apache2.4.51\\bin\\httpd.exe -d C:/wamp64/bin/apache/apache2.4.51'
[Tue May 10 20:57:12.523829 2022] [mpm_winnt:notice] [pid 4464:tid 636] AH00418: Parent: Created child process 4888
[Tue May 10 20:57:13.086802 2022] [mpm_winnt:notice] [pid 4888:tid 672] AH00354: Child: Starting 64 worker threads.
[Tue May 10 20:57:58.414874 2022] [authz_core:error] [pid 4888:tid 1196] [client 172.16.0.108:53093] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/
[Tue May 10 20:57:58.618016 2022] [authz_core:error] [pid 4888:tid 1196] [client 172.16.0.108:53093] AH01630: client denied by server configuration: C:/wamp64/www/snipe-it/public/favicon.ico, referer: http://172.16.0.115/

**Comment by uberbrady** [comment]
"Client denied by server configuration" maybe is the thing? Something in your Apache config that doesn't allow outside access?

**Reply by J1MP123** [reply]
Not sure what i have missed any idea what should i bee looking for?

**Reply by J1MP123** [reply]
Could you message me on discord HWguy#4297

**Comment by MsideSyncro** [comment]
Its possible that in your VirtualHost config that the IP requires http:// in front of the IP. What you have listed above is just the IP. I had to put http:// in both the env and this config file

<VirtualHost *:80>
ServerName 172.16.0.115
DocumentRoot "${INSTALL_DIR}/www/snipe-it/public"
<Directory "${INSTALL_DIR}/www">
Options +Indexes +Includes +FollowSymLinks +MultiViews
AllowOverride All
Require all granted
